data/pam_test/pam.sh

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env bats
+# pam-test tool from the url https://github.com/pbrezina/pam-test
+
+USER_NOR='pamtest'
+USER_NOR_PW='pamtest'
+USER_PAM_DES='tstpamunix_des'
+USER_PAM_DES_PW='0aXKZztA.d1KY'
+USER_PAM_BIG='tstpamunix_big'
+USER_PAM_BIG_PW='0aXKZztA.d1KYIuFXArmd2jU'
+USER_ERR='pamtestx'
+USER_ERR_PW='pamtestxx'
+ROOT='root'
+ROOT_PW='nots3cr3t'
+source ./pam_test.sh
+
+setup() {
+    # the dir /etc/pam.d/pam_test is static in the tool pam-test
+    cp stacks/system-default /etc/pam.d/pam_test
+    cp /etc/security/access.conf /etc/security/access.conf.bak
+    cp /etc/security/time.conf /etc/security/time.conf.bak
+    cp /etc/security/group.conf /etc/security/group.conf.bak
+    cp /etc/security/limits.conf /etc/security/limits.conf.bak
+}
+
+teardown() {
```

Comment:
You need to make sure that teardown() is always called, otherwise you could mess up the SUT. 

I suggest that this test module runs on it's separate test suite, or that it restores the system to a previous snapshot

Reply:
yes, teardown() is always called by "BATS", it make sure everything is OK.
And I agree with you that it should restore the system to a previous snapshot,
do you think it is OK to add "test_flags" function into this perl module?

Reply:
Sorry, I have a question. I don't quite understand this sentence "I suggest that this test module runs on it's separate test suite", do you mean that I need to create a new test suite like "mau-extratests" on openQA?

Reply:
@badboywj It's just a suggestion. The problem is: if the test fails, can you guarantee that BATS will always restore the files?

Reply:
> do you think it is OK to add "test_flags" function into this perl module?

Totally :)

Reply:
> @badboywj It's just a suggestion. The problem is: if the test fails, can you guarantee that BATS will always restore the files? But if we have a snapshot to return to, it might not be needed

OK. I will improve my code and have a try to make system to return to last good snapshot.

Reply:
> > @badboywj It's just a suggestion. The problem is: if the test fails, can you guarantee that BATS will always restore the files? But if we have a snapshot to return to, it might not be needed
> 
> OK. I will improve my code and have a try to make system to return to last good snapshot.

A btrfs snapshot could work too :)

Reply:
teardown() is always called by "bats", even if some of tests in the "bats" script pam.sh failed. But just in case, I used "undochange" function from "snapper" to restore all changed files, and it doesn't need to reboot.